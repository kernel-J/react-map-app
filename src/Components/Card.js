import React from 'react';
import styled from 'styled-components';
import PropTypes from 'prop-types';

const Base = styled.div`
  padding: 10;
  border: 1px solid;
  background-color: ${props => (props.selected ? 'red' : 'white' )};
`

const Label = styled.div`
  font-family: Helvetica, sans-serif;
  font-weight: bold;
  margin: 10px;
`

const Card = (props) => {
  const { onSelect, onUnselect, city } = props;
  return (
    <Base 
      onMouseEnter={onSelect}
      onMouseLeave={onUnselect}
      selected={city.selected}
    >
      <Label>City: {city.city}</Label>
      <Label>Population: {city.population}</Label>
      <Label>State: {city.state}</Label>
    </Base>
  )
}

Card.propTypes = {
  onSelect: PropTypes.func,
  onUnselect: PropTypes.func,
  city: PropTypes.object
}

export default Card;
