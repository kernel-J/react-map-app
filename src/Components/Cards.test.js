import React from 'react';
import renderer from 'react-test-renderer';

import Card from './Card';

test('render default Card', () => {
  const data = {
    selected: false,
    city: 'New York',
    population: 8405837,
    state: 'New York'
  };
  const component = renderer.create(
    <Card city={data} />
  );

  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test('render selected Card', () => {
  const data = {
    selected: true,
    city: 'New York',
    popuation: 8405837,
    state: 'New York'
  };
  const component = renderer.create(
    <Card city={data} />
  );
  
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
