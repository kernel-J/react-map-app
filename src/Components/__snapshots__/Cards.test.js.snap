// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render default Card 1`] = `
<div
  className="sc-bdVaJa jiJPRA"
  selected={false}
>
  <div
    className="sc-bwzfXH cFofpB"
  >
    City: 
    New York
  </div>
  <div
    className="sc-bwzfXH cFofpB"
  >
    Population: 
    8405837
  </div>
  <div
    className="sc-bwzfXH cFofpB"
  >
    State: 
    New York
  </div>
</div>
`;

exports[`render selected Card 1`] = `
<div
  className="sc-bdVaJa kJYvwi"
  selected={true}
>
  <div
    className="sc-bwzfXH cFofpB"
  >
    City: 
    New York
  </div>
  <div
    className="sc-bwzfXH cFofpB"
  >
    Population: 
  </div>
  <div
    className="sc-bwzfXH cFofpB"
  >
    State: 
    New York
  </div>
</div>
`;
